.github/workflows/codeql.yml

```
@@ -0,0 +1,46 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
```

Before you merge, you can temporarily add your feature-branch to this list. Pushing that commit will allow you to verify that the workflow runs OK for the `push` event -before- a merge to main
```suggestion
    branches: [ main, feat/add-codeql-workflow ]
```

---

It seems the action is running without this rule, and it was even running on push when I had `paths: 'src/**'`